packages/components/src/product-icon/index.stories.jsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+import './index.stories.scss';
+import ProductIcon from '.';
+import { supportedSlugs } from './config';
+
+export default { title: 'ProductIcon' };
+
+export const Default = () => {
+	return (
+		<>
+			{ supportedSlugs.map( slug => (
+				<div className="index.stories__icon-tile">
```

Comment:
Why does this class include a literal `.`? That seems unorthodox 😮

Reply:
The linting enforces it, I think based on the file name - `index.stories.jsx`

Reply:
In this case, the rule also allows `product-icon` (the surrounding folder name), which is a more wise choice of a CSS class name.

I filed #38764 to prevent ESLint from suggesting the less reasonable `index.stories` name.

Reply:
Are we adhering to BEM? If we are I'd argue that the `index.stories` block and elements aren't part of the `product-icon` block, particularly because `index.stories__icon` _contains_ a `product-icon` block.

While I found the rule and the dot syntax unexpected too, it makes sense to me that the storybook specific styling is seperate from the actual component styling.